praw/objector.py

```diff
@@ -1,44 +1,24 @@
 """Provides the Objector class."""
-import re
-
 from .exceptions import APIException
+from .util import snake_case_keys
 
 
 class Objector(object):
     """The objector builds :class:`.RedditBase` objects."""
 
-    @staticmethod
-    def _camel_to_snake(name):
-        """Return `name` converted from camelCase to snake_case.
-
-        Code from http://stackoverflow.com/a/1176023/.
-
-        """
-        first_break_replaced = re.sub("(.)([A-Z][a-z]+)", r"\1_\2", name)
-        return re.sub(
-            "([a-z0-9])([A-Z])", r"\1_\2", first_break_replaced
-        ).lower()
-
-    @classmethod
-    def _snake_case_keys(cls, dictionary):
-        """Return a copy of dictionary with keys converted to snake_case.
-
-        :param dictionary: The dict to be corrected.
-
-        """
-        return {cls._camel_to_snake(k): v for k, v in dictionary.items()}
-
-    def __init__(self, reddit):
+    def __init__(self, reddit, parsers=None):
         """Initialize an Objector instance.
 
         :param reddit: An instance of :class:`~.Reddit`.
 
         """
-        self.parsers = {}
+        self.parsers = {} if parsers is None else parsers
         self._reddit = reddit
 
     def kind(self, instance):
-        """Return the kind from the instance class.
+        """Deprecated.
```

Comment:
Do we need to deprecate this method, or can we simply remove it?

Reply:
I’m already planning to remove this method next PR, when I make all keys of `self.parsers` constant.

Currently, @jarhill0 actually [uses this method in his latest PR][1].

[1]: https://github.com/praw-dev/praw/pull/1067/files#diff-d0073be05410b6f42193fdeaf09218e2R233

Reply:
My use of that method was actually a mistake. Let me submit a PR to fix that.

Reply:
Done.

I’ve removed `.register()` too since that’s not going to be needed anymore either.